### PR TITLE
chore: upgrade sdk js to V3 and lambda to runtime 20.x

### LINKED ISSUE #5422: Node.js 16 is now 'End of Life'
https://nodejs.org/en/blog/announcements/nodejs16-eol

'We are moving the End-of-Life date of Node.js 16 by seven months to coincide with the end of support of OpenSSL 1.1.1 on September 11th, 2023.'

With the newest version of the copilot-cli, generated Lambdas still use Node.js 16 which is EOL.
Are there any plans to upgrade this to 18 or 20?

### PR DESCRIPTION
**Pull request by KollaAdithya**
Resolves #5422
On top of this PR #5429 upgraded sdk js to V3

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the Apache 2.0 License.

### CODE REVIEW

File: site/content/docs/manifest/pipeline.en.md [hunk 1]
```diff
@@ -17,7 +17,7 @@ List of all available properties for a Copilot pipeline manifest. To learn more
             # connection_name: a-connection
     
         build:
-          image: aws/codebuild/amazonlinux2-x86_64-standard:4.0
+          image: aws/codebuild/amazonlinux2-x86_64-standard:5.0
```
Comment (iamhopaul123) [inline site/content/docs/manifest/pipeline.en.md:20]: Do you know if the following scenarios will break for pipeline users

1. They don't update the pipeline by running `copilot pipeline deploy` again so the codebuild won't be updated. However, they updated the buildspec to use the latest Copilot.
2. They updated the pipeline and the codebuild was upgrade to `amazonlinux2-x86_64-standard:5.0`. However, they forgot to update the buildspec file.

If so, we might want to make sure they are fully aware of the reason why this happen and what to do to get it around (by looking at the codebuild log).

**Review by iamhopaul123 — COMMENTED** [review]
Good work! Thank you so much for making this happen! Do we need to update .release too?

File: site/content/docs/manifest/pipeline.en.md (hunk 1 above)
Comment (KollaAdithya) [reply site/content/docs/manifest/pipeline.en.md:20]: I did not test these scenarios. I will  deploy a pipeline to test these

Comment (KollaAdithya) [reply site/content/docs/manifest/pipeline.en.md:20]: I tested both the scenarios, surprisingly it did not error out.

Comment (iamhopaul123) [reply site/content/docs/manifest/pipeline.en.md:20]: Cool! That's even better. It seems like we don't depend on the current env nodejs when they use Copilot.

**Review by iamhopaul123 — APPROVED** [review]

File: site/content/docs/manifest/pipeline.en.md (hunk 1 above)
Comment (KollaAdithya) [reply site/content/docs/manifest/pipeline.en.md:20]: I have only default buildspec that copilot generates other than changing it to latest version.

what if the customer have some commands like `make test` where they are running some unit tests for their application based on nodejs environment.

Do we need consider this scenario?

Comment (iamhopaul123) [reply site/content/docs/manifest/pipeline.en.md:20]: Hmm it should be good i think, as they should be aware of their dependency on nodejs version. Whereas if the default scenario breaks, they probably don't know why nodejs version would even matter.

**Review by CaptainCarpensir — APPROVED** [review]

**Pull request merged**